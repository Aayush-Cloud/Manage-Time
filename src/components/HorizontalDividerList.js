// @flow
import * as React from 'react';
import { View } from 'react-native';
import styles from './styles/HorizontalDividerList.styles';

type Props = {
  children?: React.Node,
  dividerStyle?: any,
  hasBottomDivider?: boolean,
};

export default function HorizontalDividerList(props: Props) {
  const { children, dividerStyle, hasBottomDivider } = props;
  return (
    <View>
      {React.Children.toArray(children).reduce((elements, child, i, array) => {
        elements.push(child);
        if (
          (hasBottomDivider && i < array.length) ||
          (!hasBottomDivider && i < array.length - 1)
        ) {
          elements.push(
            <View
              style={[styles.divider, dividerStyle]}
              key={`${child.key} divider line`}
            />,
          );
        }
        return elements;
      }, [])}
    </View>
  );
}

HorizontalDividerList.defaultProps = {
  children: undefined,
  dividerStyle: undefined,
  hasBottomDivider: false,
};
