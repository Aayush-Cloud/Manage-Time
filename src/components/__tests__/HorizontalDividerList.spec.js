// @flow
import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import HorizontalDividerList from '../HorizontalDividerList';

describe('HorizontalDividerList component', () => {
  it('renders children with horizontal divider lines between them', () => {
    const tree = renderer.create(
      <HorizontalDividerList>
        <Text>Child 1</Text>
        <Text>Child 2</Text>
        <Text>Child 3</Text>
      </HorizontalDividerList>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
