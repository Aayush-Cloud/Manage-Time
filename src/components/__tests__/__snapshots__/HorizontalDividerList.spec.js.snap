// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalDividerList component renders children with horizontal divider lines between them 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Child 1
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.12)",
          "height": 1,
          "marginHorizontal": 0,
        },
        undefined,
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Child 2
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.12)",
          "height": 1,
          "marginHorizontal": 0,
        },
        undefined,
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Child 3
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.12)",
          "height": 1,
          "marginHorizontal": 0,
        },
        undefined,
      ]
    }
  />
</View>
`;
